src/sdltiles.cpp

```diff
@@ -255,6 +259,46 @@ void init_interface()
 //***********************************
 //Non-curses, Window functions      *
 //***********************************
+void generate_alt_rect_texture()
+{
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+    static const Uint32 rmask = 0xff000000;
+    static const Uint32 gmask = 0x00ff0000;
+    static const Uint32 bmask = 0x0000ff00;
+    static const Uint32 amask = 0x000000ff;
+#else
+    static const Uint32 rmask = 0x000000ff;
+    static const Uint32 gmask = 0x0000ff00;
+    static const Uint32 bmask = 0x00ff0000;
+    static const Uint32 amask = 0xff000000;
+#endif
+    SDL_Surface *altsurf = SDL_CreateRGBSurface( 0, 1, 1, 32, rmask, gmask, bmask, amask );
```

Comment:
Same here, use `SDL_Surface_Ptr` and than you don't need the call to `SDL_FreeSurface`.